src/screens/WaitTimeScreen.tsx

```diff
@@ -1,42 +1,471 @@
+                  onPress={() => {
+                    setIsVisible(false);
+                    setSortBy("Distance");
+                  }}
+                >
+                  <StyledText>Distance</StyledText>
+                  <View style={styles.checkIconDistance}>
+                    {sortBy === "Distance" && (
+                      <Feather name="check" size={24} color="black" />
+                    )}
+                  </View>
+                </Button>
+              </View>
+            </View>
+          </TouchableWithoutFeedback>
+        </View>
+      </TouchableWithoutFeedback>
+    </Modal>
+  );
+
+  let searchBar: SearchBar | null;
   return (
-    <View style={styles.container}>
-      <StyledText style={styles.huge}>Wait Times</StyledText>
-      <WhiteSpace />
-      <Button
-        type={"primary"}
-        onPress={() =>
-          navigation.navigate(LOCATION_DETAILS, { location: "Location X" })
-        }
-      >
-        <StyledText>Go to Location X</StyledText>
-      </Button>
-      <StatusBar style="auto" />
-    </View>
+    <TouchableWithoutFeedback
+      onPress={() => {
+        searchBar?.inputRef?.blur();
+      }}
+    >
+      <View style={theme.styles.screenContainer}>
+        <View style={styles.searchContainer}>
+          <SearchBar
+            ref={(el) => ((searchBar as any) = el)}
+            placeholderTextColor={theme.styles.placeholderText.color}
+            style={styles.searchBar}
+            value={searchValue}
+            placeholder="Search places"
+            onChange={(value) => {
+              setSearchValue(value);
+            }}
+            onSubmit={(value) => {
+              setSearchValue(value);
+            }}
+            cancelText={"Cancel"}
+            onCancel={() => {
+              searchBar?.inputRef?.blur();
+            }}
+          />
+        </View>
+        <View style={styles.sortContainer}>
+          <ScrollView horizontal={true} showsHorizontalScrollIndicator={false}>
+            <Button
+              style={styles.sortByTag}
+              onPress={() => {
+                setIsVisible(true);
+              }}
+            >
+              <StyledText style={styles.sortByText}>Sort By</StyledText>
+            </Button>
+            {poiTypes.map((type: string, index: number) => {
+              return (
+                <Tag
+                  key={index}
+                  style={styles.tags}
+                  styles={StylesOverride.tagStyles}
+                >
+                  {type}
+                </Tag>
+              );
+            })}
+          </ScrollView>
+        </View>
+        {modalComponent}
+        <View style={styles.stepsContainer}>
+          <Steps
+            styles={StylesOverride.stepsStyles}
+            size="small"
+            direction="horizontal"
+          >
+            {steps1.map((_item: any, index: any) => (
+              <Step key={index} status={"wait"} />
```

Comment:
Best practices for react suggests not using index as the key for mappings. At least, not alone.
https://www.developerway.com/posts/react-key-attribute  
https://articles.wesionary.team/using-index-as-a-key-is-an-anti-pattern-in-react-8e5db3aea3a9

Reply:
Used strings like the second link used